src/Simulation/Simulators/ResourcesEstimator/RuntimeCounter.cs

```diff
@@ -0,0 +1,62 @@
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime;
+
+namespace Simulator
+{
+    public class RuntimeCounter : IQCTraceSimulatorListener, ICallGraphStatistics
+    {
+        public RuntimeCounter()
+        {
+            AddToCallStack(CallGraphEdge.CallGraphRootHashed, OperationFunctor.Body);
+            stats = new StatisticsCollector<CallGraphEdge>(
+                new [] { "Runtime" },
+                StatisticsCollector<CallGraphEdge>.DefaultStatistics()
+            );
+        }
+
+        public bool NeedsTracingDataInQubits => false;
+
+        public object? NewTracingData(long qubitId) => null;
+
+        public void OnAllocate(object[] qubitsTraceData) {}
+
+        public void OnRelease(object[] qubitsTraceData) {}
+
+        public void OnBorrow(object[] qubitsTraceData, long newQubitsAllocated) {}
+
+        public void OnReturn(object[] qubitsTraceData, long qubitReleased) {}
+
+        public void OnOperationStart(HashedString name, OperationFunctor variant, object[] qubitsTraceData)
+        {
+            AddToCallStack(name, variant);
+            operationCallStack.Peek().Watch.Start();
+        }
+
+        public void OnOperationEnd(object[] returnedQubitsTraceData)
```

Comment:
There is no documentation in this file. At least the non-empty methods should contain some information on what they are implementing.